AGENTS.md

```diff
@@ -57,19 +57,44 @@ When using Guice modules for dependency injection:
   ```
 - ❌ **DON'T**: Use `@Named` directly on implementation classes without a corresponding `@Provides` method
   - **Why**: Violates Guice best practices; prefer explicit provider methods for clarity
-- ❌ **DON'T**: Use `bind` methods in `AbstractModules`
-  - **Why**: Violates Guice best practices; prefer explicit provider methods for clarity
 
 ### Java Language Features
 
 - ✅ **DO**: Use Java records for simple data classes
   ```java
-  public record JfrSession(String name, long id) {}
+  public record JfrSession(
+      @JsonProperty @NonNull String name,
+      @JsonProperty long id) {}
```

Comment:
Please remove the `JsonProperty` annotation. Jackson can serialize this without it.